### PR TITLE
Refactor `Stream`

### PR DESCRIPTION
**Pull request by LVala**
Requires some more changes, but the PR would get a bit too chunky.

### CODE REVIEW

File: client/src/streams/stream.rs [hunk 1]
```diff
@@ -1,122 +1,248 @@
...
-        self.previous_rtp_timestamp = Some(rtp.timestamp as f64);
+        let unit = 1.0 / clock_rate as f64;
+        let arrival_diff = rtp_info
+            .time
+            .checked_sub(prev_rtp_info.time)
+            .unwrap()
+            .as_secs_f64();
+        let rtp_timestamp_diff =
+            (rtp_info.packet.timestamp as i64 - prev_rtp_info.packet.timestamp as i64) as f64;
+        let diff = arrival_diff - rtp_timestamp_diff * unit;
```
Comment (Falon452) [inline client/src/streams/stream.rs:183]:
```suggestion
        let diff = (arrival_diff - rtp_timestamp_diff).abs() * unit;
```

File: client/src/app/gui/rtp_streams_table.rs [hunk 2]
```diff
@@ -79,61 +76,10 @@ impl RtpStreamsTable {
...
-                            format!("packet number = {}\njitter = {:.5}ms", value.x, value.y)
-                        })
-                        .set_margin_fraction(Vec2::new(0.1, 0.1))
-                        .allow_scroll(false)
-                        .show(ui, |plot_ui| {
-                            plot_ui.line(line_zero_axis);
-                            plot_ui.line(line);
-                        });
-                    ui.add_space(7.0);
-                });
```
Comment (Falon452) [inline client/src/app/gui/rtp_streams_table.rs:136]: What about this, are we deleting it?

**Review by Falon452 — APPROVED** [review]

File: client/src/app/gui/rtp_streams_table.rs [hunk 3]
```diff
@@ -79,61 +76,10 @@ impl RtpStreamsTable {
...
+                ui.label(format!("{:?}", stream.get_expected_count()));
```
Comment (Falon452) [inline client/src/app/gui/rtp_streams_table.rs:82]: Are we deleting this? If not let's add TODO

Comment (LVala) [reply client/src/app/gui/rtp_streams_table.rs:82]: Gonna take care of the Streams table view in the next PR.

File: client/src/streams/stream.rs (hunk 1 above)
Comment (LVala) [reply client/src/streams/stream.rs:183]: I don't think this should be an absolute value, but I'm going to take a closer look at the jitter when refactoring the Streams table view.

**Pull request merged**